### PR TITLE
added task solution

### PR DESCRIPTION
**Pull request by AnyaRatusnyak**

### CODE REVIEW

File: src/Ticket.java [hunk 1]
```diff
@@ -1,121 +1,159 @@
...
     }
 
-    public Ticket(
-            String concertHall,
-            int eventCode
-    ) {
-        validateAndInitialize(
-                null,
+    public Ticket(String concertHall, int eventCode) {
+        super(0);
```
Comment (Alexmansar) [inline src/Ticket.java]: why 0? yiu can just provide in costructor, as you made line 20

File: src/TicketService.java [hunk 2]
```diff
@@ -29,11 +31,30 @@ public static void main(String[] args) {
         for (Ticket ticket : ticketsBySector) {
             System.out.println(ticket);
         }
+
+        Client client = new Client(1);
```
Comment (Alexmansar) [inline src/TicketService.java]: please, use abstraction, not realisation 
User client = new Client()

File: src/TicketService.java [hunk 3]
```diff
@@ -29,11 +31,30 @@ public static void main(String[] args) {
...
+        client.printRole();
+        Ticket clientTicket = client.getTicket();
+        clientTicket.setId(5);
+        clientTicket.shared("+380971111111", "example@gmail.com");
+        clientTicket.shared("+380971111111");
+        clientTicket.print();
+
+        Admin admin = new Admin(1);
```
Comment (Alexmansar) [inline src/TicketService.java]: and here

File: src/util/AnnotationChecker.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package util;
+
+import annotation.NullableWarning;
+
+import java.lang.reflect.Field;
+
+public class AnnotationChecker {
```
Comment (Alexmansar) [inline src/util/AnnotationChecker.java]: NullableWarningValidator for example.

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@AnyaRatusnyak Hi, in general good job, just few comments))

File: src/Ticket.java (hunk 1 above)
Comment (AnyaRatusnyak) [reply src/Ticket.java]: Ok, thank you)

File: src/TicketService.java (hunk 2 above)
Comment (AnyaRatusnyak) [reply src/TicketService.java]: If I use abstraction , I can't use method getTicket() from  Client class.  Do I need to create new interface with this method ?

Comment (Alexmansar) [reply src/TicketService.java]: ((Client) client).getTicket() ; works?

Comment (AnyaRatusnyak) [reply src/TicketService.java]: Sorry! I forgot about casting (( Thank you

File: src/Ticket.java [hunk 5]
```diff
@@ -1,121 +1,159 @@
...
+        this.concertHall = concertHall;
+        this.eventCode = eventCode;
+        this.time = System.currentTimeMillis() / 1000;
+        this.isPromo = false;
+        this.weight = 0.00;
+        this.price = 0.00;
     }
 
     public Ticket() {
+        super(0);
```
Comment (Alexmansar) [inline src/Ticket.java]: super(id); :)

**Review by Alexmansar — APPROVED** [review]
@AnyaRatusnyak Good job))
please add new line in the constructor (see commnet), and merge it (without additional approve)

File: src/Ticket.java (hunk 5 above)
Comment (AnyaRatusnyak) [reply src/Ticket.java]: Thank you very much!

**Review by Alexmansar — APPROVED** [review]
@AnyaRatusnyak Hi,could be merged) Good job

**Pull request merged**